generated/nifake/session.py

```diff
@@ -113,10 +140,16 @@ class _SessionBase(object):
     An attribute of type string with read/write access.
     '''
 
-    def __init__(self, repeated_capability):
-        self._library = library_singleton.get()
+    def __init__(self, repeated_capability, vi=None, library=None, encoding=None, freeze_it=False):
```

Comment:
Why are vi, library, encoding optional?

Reply:
I think it's less error prone if they weren't optional and Session passes them when it chains up to its parent class' `__init__()` method.

Reply:
Which is preferred?
```python
super().__init__(rep_cap, None, None, None)
self._vi = ...
self._library = ...
self._encoding = ...
```

or
```python
self._vi = ...
self._library = ...
self._encoding = ...
super().__init__(rep_cap, self._vi, self._library, self._encoding)
```

Reply:
Normally it's good practice to chain up the constructor first, then add additional code.

Reply:
That is why I had made them optional. So don't make them optional, but explicitly pass in `None`?